pkg/apis/core/v1/storagecluster.go

```diff
@@ -122,6 +122,28 @@ type VolumeSpec struct {
 	MountPropagation *v1.MountPropagationMode `json:"mountPropagation,omitempty"`
 }
 
+// TLSSpec is the spec used to define TLS configuration for a cluster
+type TLSSpec struct {
+	// Defaults to parent (i.e. if missing, takes value from spec.security.enabled )
+	Enabled *bool `json:"enabled,omitempty"`
+	// AdvancedTLSOptions provides overrides to the information specified in the spec.security.tls.certSecret
+	AdvancedTLSOptions *AdvancedTLSOptions `json:"advancedOptions,omitempty"`
+}
+
+// AdvancedTLSOptions is a reference to TLS certificates
+type AdvancedTLSOptions struct {
+	APIRootCA  *CertLocation `json:"apirootCA,omitempty"`
+	ServerKey  *CertLocation `json:"certKey,omitempty"`
+	ServerCert *CertLocation `json:"serverCert,omitempty"`
+}
+
+// CertLocation specifies where portworx should pick up the certificate.
+// Certificate can be in a file on a fixed location or in a secret
+type CertLocation struct {
+	// filename on the node for the cert file. All files must be installed on a fixed location (/etc/pwx)
+	FileName *string `json:"filename,omitempty"`
```

Comment:
> Does this need to be a pointer?

Yes, because in future we will have either a k8s secret or a file as source. Something like:
```
serverCert:
  filename: <could be blank if k8s secret supplied as source>
  secretKeyRef:
    name: 
    key: 
}
```